ada-parser/src/main/java/com/ucl/ADA/parser/parser/CompilationUnitBuilder.java

```diff
@@ -0,0 +1,83 @@
+package com.ucl.ADA.parser.parser;
+
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTParser;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Map;
+
+public class CompilationUnitBuilder {
+
+    /**
+     * Default constructor for SourceFileProcessor
+     */
+    protected CompilationUnitBuilder() {
+    }
+
+    /**
+     * It makes the compilation unit for a given file.
+     *
+     * @param filePath                 : Source file path
+     * @param sourceCode               :Source content that needs to make compilation unit.
+     * @param allSrcDirectories:Source directories for configuring the bildings
+     * @return A compilation unit for the given source file.
+     */
+
```

Comment:
Please remove the whiteline between the javadoc and the corresponding method here and everywhere